src/components/SortableTable/SortableTable.tsx

```diff
@@ -0,0 +1,270 @@
+
+type HorizontalAlignment = 'left' | 'center' | 'right';
+
+export interface SortableColumn<T>
+  extends Omit<HeaderProps, 'children' | 'onSort'> {
+  align?: HorizontalAlignment;
+  cell?: (row: T, expanded?: boolean) => ReactNode;
+  footer?: ReactNode;
+  header?: ReactNode;
+  hidden?: boolean;
+  key?: Key;
+  onSort?: (ascending: boolean) => void;
+  sortable?: boolean;
+  width?: string;
+}
+
+export type SortableTableRow<T> = T & { key: Key | undefined };
+type RowClassNameFunction<T> = (row: T) => ReactNode | undefined;
+type RowExpandedFunction<T> = (row: T) => ReactNode | false;
+type RowOnClickFunction<T> = (row: T, evt: MouseEvent) => void;
+type RowSelectedFunction<T> = (row: T) => boolean;
+
+export interface SortableTableProps<T> extends Omit<TableProps, 'children'> {
+  allSelected?: boolean;
+  columns: SortableColumn<T>[];
+  expandableColumn?: Partial<SortableColumn<T>>;
+  footer?: ReactNode;
+  key?: string;
+  onExpand?: (row: T) => void;
+  onSelect?: (row: T, selected?: boolean) => void;
+  onSelectAll?: (selected: boolean) => void;
+  /* eslint-disable-next-line no-use-before-define */
+  renderRow?: typeof defaultRenderRow;
+  rowClassName?: RowClassNameFunction<T>;
+  rowExpanded?: RowExpandedFunction<T>;
+  rowOnClick?: RowOnClickFunction<T>;
+  rows?: SortableTableRow<T>[];
+  rowSelected?: RowSelectedFunction<T>;
+  truncate?: boolean;
+}
+
+function generateColumnClassName<T>(
+  column: SortableColumn<T>,
+  truncate = false
+) {
+  return classnames(
+    truncate && 'text-truncate',
+    column.align && `text-${column.align}`,
+    column.className
+  );
+}
+
+function defaultRenderRow<T>(
+  row: SortableTableRow<T>,
+  columns: SortableColumn<T>[],
+  rowClassName?: RowClassNameFunction<T>,
+  rowExpanded?: RowExpandedFunction<T>,
+  rowOnClick?: RowOnClickFunction<T>,
+  truncate?: boolean,
+  rowSelected?: RowSelectedFunction<T>
+): ReactNode {
+  const expanded = rowExpanded && rowExpanded(row);
+  return [
+    <tr
+      key={row.key}
+      className={classnames(
+        { 'table-primary': rowSelected && rowSelected(row) },
+        rowClassName ? rowClassName(row) : ''
+      )}
+      onClick={(e) => rowOnClick && rowOnClick(row, e)}
+      role={rowOnClick ? 'button' : undefined}
+    >
+      {columns.map((column) => (
+        <td
+          key={column.key}
+          className={generateColumnClassName(column, truncate)}
+        >
+          {column.cell && column.cell(row, expanded !== false)}
+        </td>
+      ))}
+    </tr>,
+    expanded && <tr key={row.key ? `${row.key}-hidden` : undefined} hidden />,
+    expanded && (
+      <tr
+        key={row.key ? `${row.key}-expanded` : undefined}
+        className={classnames(
+          { 'table-primary': rowSelected && rowSelected(row) },
+          'tr-expanded'
+        )}
+      >
+        <td className="border-top-0" colSpan={columns.length}>
+          {expanded}
+        </td>
+      </tr>
+    ),
+  ];
+}
```

Comment:
Agreed. I'm hesitant to add to the complexity of this PR.
Also, optimally, it would change the call from a function to a more composition-focused syntax:

Instead of:
```
<SortableTable renderRow={myRowRenderFn} ... />
```
do this:
```
<SortableTable row={MyRowComponent} ... />
```